packages/dvilela/connections/mirror_db/connection.py

```diff
@@ -42,30 +42,60 @@
 
 PUBLIC_ID = PublicId.from_str("dvilela/mirror_db:0.1.0")
 
+# Default headers for JSON requests
+DEFAULT_HEADERS = {"Content-Type": "application/json", "accept": "application/json"}
+
 
 def retry_with_exponential_backoff(max_retries=5, initial_delay=1, backoff_factor=2):  # type: ignore
     """Retry a function with exponential backoff."""
 
     def decorator(func):  # type: ignore
         @wraps(func)
         async def wrapper(*args, **kwargs):  # type: ignore
+            connection_instance = args[
+                0
+            ]  # Assuming the first arg is self (the connection instance)
             delay = initial_delay
             for attempt in range(max_retries):
                 try:
                     return await func(*args, **kwargs)
-                except Exception as e:
-                    if "rate limit exceeded" in str(e).lower():
+                except aiohttp.ClientResponseError as e:
+                    # Check if the error is rate limiting (e.g., status code 429)
+                    # You might need to adjust the status code based on the API
+                    if e.status == 429:
                         if attempt < max_retries - 1:
-                            print(f"Retrying in {delay} seconds due to rate limit...")
+                            connection_instance.logger.warning(
+                                f"Rate limit exceeded. Retrying in {delay} seconds..."
+                            )
                             await asyncio.sleep(delay)
                             delay *= backoff_factor
                         else:
-                            print(
-                                "Max retries reached. Could not complete the request."
+                            connection_instance.logger.error(
+                                "Max retries reached for rate limiting. Could not complete the request."
```

Comment:
Please fix this nesting. It's the same thing we discussed the other day. Rule of thumb: avoid using "else". That forces you to rearrange and simplify your conditionals.